updater/lib/dependabot/file_fetcher_command.rb

```diff
@@ -64,6 +72,25 @@ def save_job_details
                                        ))
     end
 
+    def dependency_files_for_multi_directories
+      @dependency_files_for_multi_directories ||= job.source.directories.map do |dir|
+        retries = 0
+        begin
+          updated_dir_source = job.source.clone.tap { |s| s.directory = dir }
+          ff = FileFetchers.for_package_manager(job.package_manager).new(
+            source: updated_dir_source,
+            credentials: Environment.job_definition.fetch("credentials", []),
+            options: job.experiments
+          )
+          ff.files
+        rescue Octokit::BadGateway => e
+          retries += 1
+          retry if retries <= 2
+          raise e
+        end
```

Comment:
Sounds like retry logic is duplicated with `dependency_files`. Perhaps you can extract a `with_retries` helper that yields to the block it receives, and rescues and retries on BadGateway errors.